data-prepper-plugins/obfuscate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/obfuscation/ObfuscationProcessor.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.obfuscation;
+
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.plugin.InvalidPluginConfigurationException;
+import org.opensearch.dataprepper.model.processor.AbstractProcessor;
+import org.opensearch.dataprepper.model.processor.Processor;
+import org.opensearch.dataprepper.model.record.Record;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import java.util.stream.IntStream;
+
+@DataPrepperPlugin(name = "obfuscate", pluginType = Processor.class, pluginConfigurationType = ObfuscationProcessorConfig.class)
+public class ObfuscationProcessor extends AbstractProcessor<Record<Event>, Record<Event>> {
+
+    public static final String COMMON_PATTERN_FORMAT = "^%\\{([A-Z_]+)}$";
+    private static final Logger LOG = LoggerFactory.getLogger(ObfuscationProcessor.class);
+    private final String key;
+    private final String newKey;
+    private final int unObfuscatedLength;
+    private final String baseReplacement;
+    private final String pattern;
+
+
+    enum CommonPattern {
```

Comment:
I'm not sure if we should retain a list of common patterns, or perhaps we should lower the number of patterns. Users will use these for important privacy-related features. I'm not sure we want to convey that we have the right answers here since we don't have the best domain experience on these.

Also, email addresses are notoriously difficult to cover with a regex. Indeed this one does not account for some of the Unicode characters which may occur.

If we do provide common patterns, we should probably err on the side of being overly aggressive. But, that is also an important discussion point we could have.

----

If we do keep these, you can move this enum to a separate file to reduce changes needed here as patterns change.